io/json/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Set the CMake minimum version.
+cmake_minimum_required( VERSION 3.13 )
+
+# Set up the library.
+detray_add_library( detray_io_json io_json
+  "include/detray/io/json_io.hpp" )
+target_link_libraries( detray_io_json INTERFACE nlohmann_json vecmem::core )
```

Comment:
You will need to use the name `nlohmann_json::nlohmann_json` here instead of simply `nlohmann_json`.

While building that external as part of the Detray build, both of those names are available.

https://github.com/nlohmann/json/blob/develop/CMakeLists.txt#L85-L86

But if we pick up an existing installation of the external with `find_package(nlohmann_json)`, only the `nlohmann_json::nlohmann_json` name will be available.

https://github.com/nlohmann/json/blob/develop/CMakeLists.txt#L171

(They really over-complicated a relatively simple setup by using way too many "helper variables" in their configuration...)